packages/cli/src/load-nodes-and-credentials.ts

```diff
@@ -216,6 +216,11 @@ export class LoadNodesAndCredentials {
 	async unloadPackage(packageName: string) {
 		if (packageName in this.loaders) {
 			this.loaders[packageName].reset();
+
+			if (this.loaders[packageName] instanceof PackageDirectoryLoader) {
```

Comment:
I think we should update `reset` to do all of this for all loaders.
let me send you a quick PR.